feat: Add exporter support for `resourcePropertyName` in `Authorization` record

ce-dmelnych opened the pull request:
### Description
Extends the exporter infrastructure to support the new `resourcePropertyName` field in `AuthorizationRecord`, enabling property-based authorization data to be properly persisted in secondary storage (RDBMS, ES/OS).

This PR implements support for the `resourcePropertyName` field across:
- **ES/OS**: Updated `AuthorizationEntity` model and authorization handlers (`AuthorizationCreatedUpdatedHandler`, `AuthorizationDeletedHandler`)
- **RDBMS**: Updated `AuthorizationDbModel` and `AuthorizationExportHandler`

Additionally, this PR adds backend support for sorting and filtering by `resourcePropertyName`, consistent with existing properties like `resourceId`. 
NOTE: that exposing these capabilities through the Authorization REST API and CamundaClient will be handled in the follow-up issue: #40079.

### Motivation
The `AuthorizationRecord` was extended with `resourcePropertyName` to enable fine-grained authorization checks at the property level. Without exporter support, this field would not be persisted to Elasticsearch/OpenSearch or RDBMS, resulting in incomplete authorization data in secondary storage.

### Changes
- Updated Elasticsearch/OpenSearch exporter components
- Updated RDBMS exporter components  
- Added support for sorting and filtering by `resourcePropertyName(s)`
- Adjusted/added new tests

## Related issues

closes #40116

ce-dmelnych committed:
feat: add resourcePropertyName to AuthorizationEntity

Extended the AuthorizationEntity models to include a new `resourcePropertyName` field.

ce-dmelnych committed:
test: fix test after extending AuthorizationEntity models

Adjusted tests to properly initialize models updated in the previous commits
by providing a value for new `resourcePropertyName` field.

ce-dmelnych committed:
feat: add RESOURCE_PROPERTY_NAME column to AUTHORIZATIONS table (rdbms)

Introduces a new RESOURCE_PROPERTY_NAME column to the AUTHORIZATIONS table.

All existing records will automatically receive an empty string ("") in this
new column during migration, ensuring backward compatibility and preventing
NULL values.

ce-dmelnych committed:
feat: support resource property name (rdbms)

ce-dmelnych committed:
feat: update SQL queries to support resourcePropertyName (rdbms)

Modified SQL queries to insert and select the new `resourcePropertyName` property
from the AUTHORIZATIONS table.
This enables persistence and retrieval of property-level authorization data through the RDBMS layer.

Support for filtering by `resourcePropertyName` will be added in a follow-up commits.

ce-dmelnych committed:
feat: add resourcePropertyName field to authorization indices (ES/OS)

Add resourcePropertyName field to Elasticsearch and OpenSearch authorization
index mappings to support exporting property-based authorization records.

ce-dmelnych committed:
feat: support sorting authorization records by the resourcePropertyName

Add support for sorting authorization records by the resourcePropertyName field
in the search query transformer (ES/OS).

Add `RESOURCE_PROPERTY_NAME("resourcePropertyName")` value in `AuthorizationSearchColumn` enum
to suport search by new property (RDBMS).

ce-dmelnych committed:
feat: add resourcePropertyName to authorization handlers (ES/OS)

Update authorization export handlers to persist the new resourcePropertyName
field from AuthorizationRecordValue to AuthorizationEntity.

ce-dmelnych committed:
feat: support filtering by `resourcePropertyName`

Added `resourcePropertyName` filtering support to AuthorizationFilter and
implemented filtering across all storages (RDBMS, ES/OS).

This enables filtering authorization records by resource property names in exported data.

The filtering support currently is not yet exposed to the REST API/CamundaClient API,
that will be implemented separately (#40079).

fabiopaini-camunda reviewed (commented):
❓ In #40805 you added a test in `AuthorizationIntegrationTest` too, why not here?

ce-dmelnych commented on security/security-services/src/test/java/io/camunda/security/impl/AuthorizationCheckerTest.java:
```diff
@@ -781,6 +788,7 @@ private AuthorizationEntity authEntity(
           resourceType.name(),
           matcher.value(),
           resourceId,
+          "",
```
Yeah, thanks for pointing on this, missed this part during compilation errors fixes.
Applied your suggestion 🙇🏻‍♂️

npepinpe reviewed (approved):
🚀 From the Identity side this is all fine

💭 I'm not super familiar with the RDBMS stuff, so it might be worth getting someone from data layer or the RDBMS project to review it. Or at least, not rely purely on me :)

🔧 We might want to also get someone to confirm the backwards compatibility here. I don't think our tests are quite there unfortunately to guarantee this :(

ce-dmelnych commented:
> 🚀 From the Identity side this is all fine
> 
> 💭 I'm not super familiar with the RDBMS stuff, so it might be worth getting someone from data layer or the RDBMS project to review it. Or at least, not rely purely on me :)
> 
> 🔧 We might want to also get someone to confirm the backwards compatibility here. I don't think our tests are quite there unfortunately to guarantee this :(

Thanks a lot, @npepinpe!
Do you have anyone in mind who could help with review of the backwards compatibility aspects?

ce-dmelnych commented:
Looping in @camunda/data-layer for a review from your side.  

This PR introduces new additions to the RDBMS table [structure](https://github.com/camunda/camunda/pull/40353/files#diff-e66e70df05eeb3cc91d208ef1e7e433e30f92a6a9f30806a2aa1edae2a228b29) and [queries](https://github.com/camunda/camunda/pull/40353/files#diff-42389b054f08dc2ad9bd50fea78c945d251763225ac182b345d942f6778c510d).  

Could you please take a look to confirm that these changes align with our existing patterns and don’t introduce any issues?

stoerti reviewed (changes requested):
I had a remark for the liquibase changeset. Please always change the original 8.9.0.xml changesets and do not create migration changesets (latest with 8.9.0 we will "lock" this file, then we need migraton changesets).

ce-dmelnych committed:
feat: inline RESOURCE_PROPERTY_NAME into AUTHORIZATIONS change set

Remove dedicated `add_resource_property_name_to_authorizations_table` change set and
add `RESOURCE_PROPERTY_NAME` column directly to original `create_authorization_permission_table` change set.

This is safe since RDBMS was not an official feature in 8.8 and incremental Liquibase files are not yet enforced.
Dropped the default value for the new column as it is now part of the initial table creation.

ce-dmelnych commented on db/rdbms-schema/src/main/resources/db/changelog/rdbms-exporter/changesets/8.9.0.xml:
Thank you, @stoerti, for the clear explanation and context 🙏🏻!

Based on your feedback, I’ve removed the separate change set and incorporated the new column definition directly into the initial change set as suggested.

stoerti reviewed (approved):
Thanks for the change, LGTM now 👍
